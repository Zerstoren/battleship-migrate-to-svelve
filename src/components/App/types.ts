import { ILobbyStore } from '../../stores/lobby';

export enum GameStatus {
  MAIN = 'main',
  LOBBY_LIST = 'list',
  CREATE_LOBBY = 'create',
  WAIT_CONNECT = 'wait',
  SET_SHIPS = 'set_ships',
  GAME = 'game_process',
  GAMEOVER = 'game_loose',
  GAMEWIN = 'game_win',
}

export enum FireTurn {
  NOBODY = 'nobody',
  ME = 'me',
  OPPONENT = 'opponent',
}

export interface IGameState {
  gameStatus: GameStatus,
  gameWithUser: boolean,

  setGameStatus: (status: GameStatus) => void,
}

export interface IProps {
  handleChangeGameStatus: (startGame: GameStatus, lobby?: ILobbyStore | null) => void,
}

export interface ISetLobbyProps {
  handleSetLobby: (startGame: GameStatus, lobby: ILobbyStore) => void
}
