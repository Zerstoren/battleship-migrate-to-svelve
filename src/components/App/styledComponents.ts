import styled from 'styled-components';

export const CenteredDiv = styled.div`
  width: 600px;
  margin: 0 auto;
  text-align: center;
`;

export const AlertError = styled.div.attrs(() => ({
  className: 'alert alert-warning alert-dismissible fade show',
}))`
text-align: center;
`;
