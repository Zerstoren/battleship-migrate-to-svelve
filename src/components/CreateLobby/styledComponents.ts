import styled from 'styled-components';

export const BlockSpacing = styled.div`
padding: 10px 0;
`;

export const FormFloating = styled.div.attrs(() => ({
  className: 'form-floating',
}))``;

export const FloatingInput = styled.input.attrs((props) => ({
  className: `form-control ${props.className as string}`,
}))``;
