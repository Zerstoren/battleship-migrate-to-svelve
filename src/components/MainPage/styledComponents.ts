import styled from 'styled-components';

export const ButtonSearchUser = styled.button.attrs(() => ({
  type: 'button',
  className: 'btn btn-secondary',
}))`
  margin: 0 12px;
`;

export const ButtonCreateLobby = styled(ButtonSearchUser)``;
