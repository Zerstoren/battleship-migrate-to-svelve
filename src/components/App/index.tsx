import { inject, observer } from 'mobx-react';
import React, { FC } from 'react';
import { ILobbyStore } from '../../stores/lobby';
import { IMainStore } from '../../stores/mainStore';
import CreateLobby from '../CreateLobby';
import EndGameScreen from '../EndGameScreen';
import Game from '../Game';
import LobbyList from '../LobbyList';
import Main from '../MainPage';
import SetShips from '../SetShips';
import WaitConnect from '../WaitConnect';
import useWebsocketAppReconnect from './hooks';
import { AlertError, CenteredDiv } from './styledComponents';
import { GameStatus } from './types';

interface IAppProps {
  mainStore?: IMainStore
}

const App: FC<IAppProps> = inject('mainStore')(observer((props) => {
  const state = props.mainStore as IMainStore;

  useWebsocketAppReconnect(state);

  const handleGameStart = (startGame: GameStatus) => state?.setGameStatus(startGame);
  const handleSetLobby = (startGame: GameStatus, lobby: ILobbyStore) => state?.setLobby(startGame, lobby);
  const handleWaitLobby = (
    startGame: GameStatus,
    lobby: ILobbyStore | null = null,
  ) => state.setLobby(startGame, lobby);

  const handleAcceptError = () => state?.setError('');

  let component = null;

  switch (state.status) {
    case GameStatus.MAIN:
      component = (<Main handleChangeGameStatus={handleGameStart} />);
      break;

    case GameStatus.CREATE_LOBBY:
      component = (<CreateLobby handleChangeGameStatus={handleWaitLobby} />);
      break;

    case GameStatus.WAIT_CONNECT:
      component = (<WaitConnect />);
      break;

    case GameStatus.LOBBY_LIST:
      component = (<LobbyList handleSetLobby={handleSetLobby} />);
      break;

    case GameStatus.SET_SHIPS:
      component = (<SetShips />); break;

    case GameStatus.GAME:
      component = (<Game />);
      break;

    case GameStatus.GAMEWIN:
    case GameStatus.GAMEOVER:
      component = (<EndGameScreen gameResult={state.status as GameStatus} />);
      break;

    default:
      break;
  }

  let err = null;

  if (state?.error) {
    err = (
      <AlertError>
        <strong>{state?.error}</strong>
        <input type="button" className="btn-close" onClick={handleAcceptError} />
      </AlertError>
    );
  }

  return (
    <>
      {err}
      <CenteredDiv>
        {component}
      </CenteredDiv>
    </>
  );
}));

export default App;
