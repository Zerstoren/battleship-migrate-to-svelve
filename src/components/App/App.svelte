<script lang="ts">
import { getContext } from 'svelte';
import type { IStore } from '../../stores';
import { STORE_CONTEXT } from '../../stores';
import CreateLobby from '../CreateLobby/CreateLobby.svelte';
import LobbyList from '../LobbyList/LobbyList.svelte';
import MainPage from '../MainPage/MainPage.svelte';
import SetShips from '../SetShips/SetShips.svelte';
import WaitConnect from '../WaitConnect/WaitConnect.svelte';
import { GameStatus } from './types';

const mainStore = getContext<IStore>(STORE_CONTEXT).mainStore;
let store = mainStore.store;

const handleAcceptError = () => {
  mainStore.setError('');
}

</script>

<div>
  {#if $store.error}
    <div class="alertError alert alert-warning alert-dismissible fade show">
      <strong>{$store.error}</strong>
      <input type="button" class="btn-close" onClick={handleAcceptError} />
    </div>
  {/if}
  <div class="centerDiv">
    {#if $store.status === GameStatus.MAIN}
      <MainPage />
    {:else if $store.status === GameStatus.CREATE_LOBBY}
      <CreateLobby />
    {:else if $store.status === GameStatus.WAIT_CONNECT}
      <WaitConnect />
    {:else if $store.status === GameStatus.LOBBY_LIST}
      <LobbyList />
    {:else if $store.status === GameStatus.SET_SHIPS}
      <SetShips />
    {/if}
  </div>
</div>

<style>
.centerDiv {
  width: 600px;
  margin: 0 auto;
  text-align: center;
}

.alertError {
  text-align: center;
}
</style>