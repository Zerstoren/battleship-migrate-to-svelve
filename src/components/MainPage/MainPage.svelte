<script lang="ts">
import { getContext } from "svelte";
import { get } from "svelte/store";

import Header from "../../shared/StyledComponents/Header.svelte";
import { STORE_CONTEXT } from "../../stores";
import type { IStore } from "../../stores";
import { GameStatus } from "../App/types";

const mainStore = getContext<IStore>(STORE_CONTEXT).mainStore;
let store = get(mainStore.store);

const onClickSearchOpponent = () => {
  mainStore.setGameStatus(GameStatus.LOBBY_LIST);
}

const onClickCreateLobby = () => {
  mainStore.setGameStatus(GameStatus.CREATE_LOBBY);
}

</script>

<Header>Battleship</Header>
<div>
  <button class="btn btn-secondary" on:click={onClickSearchOpponent}>Search opponent</button>
  <button class="btn btn-secondary" on:click={onClickCreateLobby}>Create lobby</button>
</div>

<style>
  button {
    margin: 0 12px;
  }
</style>