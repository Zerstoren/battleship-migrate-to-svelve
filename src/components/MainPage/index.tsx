import React, { FC } from 'react';
import { AppHeader } from '../../shared/StyledComponents/Headers';
import { GameStatus, IProps } from '../App/types';
import { ButtonCreateLobby, ButtonSearchUser } from './styledComponents';

const Main: FC<IProps> = ({ handleChangeGameStatus: handleStartGame } : IProps) => (
  <>
    <AppHeader>Battleship</AppHeader>
    <div>
      <ButtonSearchUser onClick={() => handleStartGame(GameStatus.LOBBY_LIST)}>
        Search opponent
      </ButtonSearchUser>

      <ButtonCreateLobby onClick={() => handleStartGame(GameStatus.CREATE_LOBBY)}>
        Create lobby
      </ButtonCreateLobby>
    </div>
  </>
);

export default Main;
