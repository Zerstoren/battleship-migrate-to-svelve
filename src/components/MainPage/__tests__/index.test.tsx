import React from 'react';
import { mount } from "enzyme";
import MainPage from '..';
import { ButtonCreateLobby, ButtonSearchUser } from '../styledComponents';
import { GameStatus } from '../../App/types';

it('render main page', () => {
  const tree = mount(<MainPage handleChangeGameStatus={() => {}} />)
  expect(tree.debug()).toMatchSnapshot();
});

it('check create lobby', () => {
  let gGameStatus = null;
  const handleGameStatus = (gameStatus: GameStatus) => gGameStatus = gameStatus;

  const wrapper = mount(<MainPage handleChangeGameStatus={handleGameStatus} />);
  wrapper.find(ButtonCreateLobby).simulate('click');
  expect(gGameStatus).toEqual(GameStatus.CREATE_LOBBY);
});

it('check search lobby', () => {
  let gGameStatus = null;
  const handleGameStatus = (gameStatus: GameStatus) => gGameStatus = gameStatus;

  const wrapper = mount(<MainPage handleChangeGameStatus={handleGameStatus} />);
  wrapper.find(ButtonSearchUser).simulate('click');
  expect(gGameStatus).toEqual(GameStatus.LOBBY_LIST);
});
