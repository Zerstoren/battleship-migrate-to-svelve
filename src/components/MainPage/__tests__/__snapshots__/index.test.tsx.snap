// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render main page 1`] = `
"<Main handleChangeGameStatus={[Function: handleChangeGameStatus]}>
  <styled.h1>
    <h1 className=\\"sc-bdfBwQ display-3\\">
      Battleship
    </h1>
  </styled.h1>
  <div>
    <styled.button onClick={[Function: onClick]}>
      <button onClick={[Function: onClick]} type=\\"button\\" className=\\"sc-gsTCUz fGVasU btn btn-secondary\\">
        Search opponent
      </button>
    </styled.button>
    <Styled(styled.button) onClick={[Function: onClick]}>
      <button onClick={[Function: onClick]} type=\\"button\\" className=\\"sc-gsTCUz sc-dlfnbm fGVasU btn btn-secondary\\">
        Create lobby
      </button>
    </Styled(styled.button)>
  </div>
</Main>"
`;
